lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1089,13 +1089,25 @@ InitKindRequest::evaluate(Evaluator &evaluator, ConstructorDecl *decl) const {
     // If we implement the ability for extensions defined in the same module
     // (or the same file) to add vtable entries, we can re-evaluate this
     // restriction.
-    if (dyn_cast<ClassDecl>(nominal) &&
-        !decl->isSynthesized() && isa<ExtensionDecl>(decl->getDeclContext()) &&
+    if (isa<ClassDecl>(nominal) && !decl->isSynthesized() &&
+        isa<ExtensionDecl>(decl->getDeclContext()) &&
         !(decl->getAttrs().hasAttribute<DynamicReplacementAttr>())) {
-      diags.diagnose(decl->getLoc(), diag::designated_init_in_extension,
-                     nominal->getName())
-        .fixItInsert(decl->getLoc(), "convenience ");
-      return CtorInitializerKind::Convenience;
+      auto classDecl = cast<ClassDecl>(nominal);
+
+      if (classDecl->getForeignClassKind() == ClassDecl::ForeignKind::CFType) {
```

Comment:
Nit: You can inline the cast here since you don't use `classDecl` after the check.
```suggestion
      if (cast<ClassDecl>(nominal)->getForeignClassKind() == ClassDecl::ForeignKind::CFType) {
```

Reply:
Done!.